src/main/scala/com/gu/scanamo/Table.scala

```diff
@@ -0,0 +1,126 @@
+package com.gu.scanamo
+
+import cats.data.Xor
+import com.gu.scanamo.error.DynamoReadError
+import com.gu.scanamo.ops.ScanamoOps
+import com.gu.scanamo.query.{Query, UniqueKey, UniqueKeys}
+
+/**
+  * Represents a DynamoDB table that operations can be performed against
+  *
+  * {{{
+  * >>> case class Transport(mode: String, line: String)
+  * >>> val transport = Table[Transport]("transport")
+  *
+  * >>> val client = LocalDynamoDB.client()
+  * >>> import com.amazonaws.services.dynamodbv2.model.ScalarAttributeType._
+  *
+  * >>> LocalDynamoDB.withTable(client)("transport")('mode -> S, 'line -> S) {
+  * ...   import com.gu.scanamo.syntax._
+  * ...   val operations = for {
+  * ...     _ <- transport.putAll(List(
+  * ...       Transport("Underground", "Circle"),
+  * ...       Transport("Underground", "Metropolitan"),
+  * ...       Transport("Underground", "Central")))
+  * ...     results <- transport.query('mode -> "Underground" and ('line beginsWith "C"))
+  * ...   } yield results.toList
+  * ...   Scanamo.exec(client)(operations)
+  * ... }
+  * List(Right(Transport(Underground,Central)), Right(Transport(Underground,Circle)))
+  * }}}
+  */
+case class Table[V: DynamoFormat](name: String) {
+  /**
+    * A secondary index on the table which can be scanned, or queried against
+    *
+    * {{{
+    * >>> case class Transport(mode: String, line: String, colour: String)
+    * >>> val transport = Table[Transport]("transport")
+    *
+    * >>> val client = LocalDynamoDB.client()
+    * >>> import com.amazonaws.services.dynamodbv2.model.ScalarAttributeType._
+    * >>> import com.gu.scanamo.syntax._
+    *
+    * >>> LocalDynamoDB.withTableWithSecondaryIndex(client)("transport", "colour-index")('mode -> S, 'line -> S)('colour -> S) {
+    * ...   val operations = for {
+    * ...     _ <- transport.putAll(List(
+    * ...       Transport("Underground", "Circle", "Yellow"),
+    * ...       Transport("Underground", "Metropolitan", "Maroon"),
+    * ...       Transport("Underground", "Central", "Red")))
+    * ...     maroonLine <- transport.index("colour-index").query('colour -> "Maroon")
+    * ...   } yield maroonLine.toList
+    * ...   Scanamo.exec(client)(operations)
+    * ... }
+    * List(Right(Transport(Underground,Metropolitan,Maroon)))
+    * }}}
+    */
+  def index(indexName: String) = Index[V](name, indexName)
+
+  def put(v: V) = ScanamoFree.put(name)(v)
+  def putAll(vs: List[V]) = ScanamoFree.putAll(name)(vs)
+  def get(key: UniqueKey[_]) = ScanamoFree.get[V](name)(key)
+  def getAll(keys: UniqueKeys[_]) = ScanamoFree.getAll[V](name)(keys)
+  def delete(key: UniqueKey[_]) = ScanamoFree.delete(name)(key)
+}
+
+private[scanamo] case class Index[V: DynamoFormat](tableName: String, indexName: String)
```

Comment:
Ah ok, I was thinking of the docstring tests as the public-facing instructions but you're just using them for all the tests. This is reasonable then!